### PR TITLE
Replace DappError with JsonRpcError #6456

### LINKED ISSUE #6456: Replace DappError with JsonRpcError
**reason why this needed:** 

some dapps like uniswap, want proper error code when error has generated, e.g it wants error code `-32603` when method `wallet_switchEthereumChain` called, and chain not supported on client.

### PR DESCRIPTION
**Pull request by oa-s**
Closes #6456

### CODE REVIEW

**Review by hboon — APPROVED** [review]
Does this require this to be included? — https://github.com/AlphaWallet/AlphaWallet-web3-provider/issues/38

**Comment by hboon** [comment]
(of course it does, you mentioned it in the issue. Hah!). So marking as `WIP. Don't Merge` first.

**Review by hboon — CHANGES_REQUESTED** [review]
Just in case. Some questions

File: AlphaWallet/Browser/ViewControllers/BrowserViewController.swift [hunk 1]
```diff
@@ -149,7 +149,7 @@ final class BrowserViewController: UIViewController {
         webView.load(URLRequest(url: url))
     }
 
-    func notifyFinish(callbackID: Int, value: Swift.Result<DappCallback, DAppError>) {
+    func notifyFinish(callbackID: Int, value: Swift.Result<DappCallback, JsonRpcError>) {
```
Comment (hboon) [inline AlphaWallet/Browser/ViewControllers/BrowserViewController.swift:152]: A few lines below this:

```
return "executeCallback(\(callbackID), \"\(error.message)\", null)"
```

Does it need to be changed? If it's not, the new code `-32603` is not passed along right?

**Review by hboon — COMMENTED** [review]
Do we need to look at https://github.com/AlphaWallet/alpha-wallet-ios/pull/6457 and https://github.com/AlphaWallet/AlphaWallet-web3-provider/pull/39 together and merge together? (then https://github.com/AlphaWallet/alpha-wallet-ios/pull/6457 needs to be updated to use the updated JS after https://github.com/AlphaWallet/AlphaWallet-web3-provider/pull/39  is merged)

**Comment by oa-s** [comment]
> Does this require this to be included? — [AlphaWallet/AlphaWallet-web3-provider#38](https://github.com/AlphaWallet/AlphaWallet-web3-provider/issues/38)

not now, after https://github.com/AlphaWallet/AlphaWallet-web3-provider/issues/38 we can update code to apply error `code`

**Review by hboon — APPROVED** [review]

**Pull request merged**